File: MonoScripting/Api/EngineClass/EntityComponents/ColliderComponent.cs
using System;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;

public class ColliderComponent : IEntityComponent {
    public ColliderComponent(ulong handle) : base(handle)
    {
    }

    Vec3 GetSize(){
        return i_GetSize(handle);
    }

    Vec3 GetCenter(){
        return i_GetCenter(handle);
    }

    void SetSize(Vec3 size){
        i_SetSize(handle, size);
    }

    void SetCenter(Vec3 center){
        i_SetCenter(handle, center);
    }

    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern Vec3 i_GetSize(ulong handle);
    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern Vec3 i_GetCenter(ulong handle);
    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern void i_SetSize(ulong handle, Vec3 size);
    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern void i_SetCenter(ulong handle, Vec3 center);
}

File: Engine/MonoScripting/Api/EntityComponents/ColliderComponent.cpp
#include "../ApiBindings.hpp"
#include <Scene/EntityComponents/ColliderComponent.hpp>
#include <MonoScripting/MonoScriptingLayer.hpp>

using namespace VSGE;

static Vec3 IsTrigger(ColliderComponent* cc) {
	return cc->IsTrigger();
}
static void SetTrigger(ColliderComponent* cc, bool trigger) {
	cc->SetTrigger(trigger);
}

static Vec3 GetSize(ColliderComponent* cc) {
	return cc->GetSize();
}
static void SetSize(ColliderComponent* cc, Vec3 size) {
	cc->SetSize(size);
}

static Vec3 GetCenter(ColliderComponent* cc) {
	return cc->GetCenter();
}
static void SetCenter(ColliderComponent* cc, Vec3 center) {
	cc->SetCenter(center);
}

void VSGE::BindCollider() {
	MonoScriptingLayer::AddInternalCall("ColliderComponent::i_IsTrigger(ulong)", IsTrigger);
	MonoScriptingLayer::AddInternalCall("ColliderComponent::i_SetTrigger(ulong,bool)", SetTrigger);

	MonoScriptingLayer::AddInternalCall("ColliderComponent::i_GetSize(ulong)", GetSize);
	MonoScriptingLayer::AddInternalCall("ColliderComponent::i_SetSize(ulong,Vec3)", SetSize);

	MonoScriptingLayer::AddInternalCall("ColliderComponent::i_GetCenter(ulong)", GetCenter);
	MonoScriptingLayer::AddInternalCall("ColliderComponent::i_SetCenter(ulong,Vec3)", SetCenter);
}